test/bls_test.js

```diff
@@ -1,146 +1,110 @@
-const bls = require("../")
-const mcl = require("mcl-wasm")
 const assert = require("chai").assert
+const keccak256 = require("keccak256")
+const mcl = require("mcl-wasm")
+
+const bls = require("../")
 
 describe("bls", () => {
 	before(async() => {
-		await mcl.init(mcl.BLS12_381)
+		await bls.init()
 	})
 
-	it("should create base point for G1", () => {
-        let q = bls.Q()
-        assert(!q.isZero(), "did not create G1 base point")
+	it("should create generator for G1", () => {
+    const g = bls.g1()
+    assert(!g.isZero(), "did not create G1 generator")
 	})
 
 	it("should hash message to G2", () => {
-		let domain = 0
-		let message = '0x6d657373616765'
-		console.log(bls.hash_to_G2(message, domain))
+		const domain = 0
+		const message = keccak256(Buffer.from('6d657373616765', 'hex'))
+		assert(bls.hashToG2(message, domain))
 	})
 
 	it("should generate a secret key", () => {
-        let s = bls.gen_secret()
-        console.log(s.getStr(10))
-        assert(s instanceof mcl.Fr && !s.isZero())
+    const s = bls.genSecret()
+    assert(s.length === 32 && !s.equals(Buffer.alloc(32)))
 	})
 
 	it("should get the corresponding public key to a secret key", () => {
-        let s = bls.gen_secret()
-        let P = bls.gen_public(s)
-        assert(s instanceof mcl.Fr)
-        assert(P instanceof mcl.G1)
-	})
-
-	it("should turn a G1 point to a public key hex string", () => {
-	    let s = bls.gen_secret()
-        let P = bls.gen_public(s)	
-		let pubstr = bls.getHexStr(P)
-		assert(pubstr.length === 98, "wrong length")
+    const s = bls.genSecret()
+    const P = bls.genPublic(s)
+    assert.equal(s.length, 32)
+    assert.equal(P.length, 48)
 	})
 
 	it("should get deserialize a hex string to a secret key and the corresponding public key", () => {
-		let secret = '263dbd792f5b1be47ed85f8938c0f29586af0d3ac7b977f21c278fe1462040e3'
-		let public = '0x0491d1b0ecd9bb917989f0e74f0dea0422eac4a873e5e2644f368dffb9a6e20fd6e10c1b77654d067c0618f6e5a7f79a'
-		let s = new mcl.Fr()
-        s.setStr(secret, 16)
-        assert(s instanceof mcl.Fr)
-        assert(s.getStr(16) === secret)
-
-        let P = bls.gen_public(s)
-        let Pstr = bls.getHexStr(P)
-
-        assert(public === Pstr, "incorrect deserialization")
-        assert(Pstr.length === 98, "wrong length")
-	})
+		const secret = '263dbd792f5b1be47ed85f8938c0f29586af0d3ac7b977f21c278fe1462040e3'
+		const pub = '9af7a7e5f618067c064d65771b0ce1d60fe2a6b9ff8d364f64e2e573a8c4ea2204ea0d4fe7f0897991bbd9ecb0d19104'
```

Comment:
why is the pubkey changed here? this test was from the test vectors https://github.com/ethereum/eth2.0-tests/blob/master/bls/test_bls.yml pls

Reply:
I didn't know where they came from and thought they were not needed :disappointed: 
I'll add them back in.

Reply:
no worries! I probably should have made a comment lol.